libtelemetry/src/main/java/com/mapbox/android/telemetry/TelemetryClient.java

```diff
@@ -184,4 +230,56 @@ private RequestBody reverseMultiForm(MultipartBody.Builder builder) {
 
     return builder.build();
   }
+
+  private MetricEvent buildMetricEvent() {
```

Comment:
i'm confused by how responsibilities are divided between classes that are sharing counters logic. can you draw a quick class diagram? perhaps, it will help you see what't the problem is.